xml/src/main/scala/akka/stream/alpakka/xml/model.scala

```diff
@@ -48,6 +48,11 @@ private class MapOverTraversable[A, K, V](source: Traversable[A], fKey: A => K,
 
   override def iterator: Iterator[(K, V)] = source.toIterator.map(a => (fKey(a), fValue(a)))
 
+  def remove(key: K): scala.collection.immutable.Map[K, V] =
+    throw new UnsupportedOperationException("Convert to map from iterator to support removing elements.")
+
+  override def updated[V1 >: V](key: K, value: V1): scala.collection.immutable.Map[K, V1] =
+    throw new UnsupportedOperationException("Convert to map from iterator to support updating elements.")
```

Comment:
Here we'd have to actually convert all elements of the traversable... perhaps indeed best to not support that since it might be surprising performance-wise?

Reply:
I decided to remove `MapOverTraversable` and go with collection conversions. Favoring no exceptions rather than slightly faster implementation.